ebean-migration/src/main/java/io/ebean/migration/runner/MigrationTable.java

```diff
@@ -296,7 +296,10 @@ private boolean shouldRun(LocalMigrationResource localVersion, LocalMigrationRes
   private boolean runMigration(LocalMigrationResource local, MigrationMetaRow existing) throws SQLException {
     String script = null;
     int checksum;
-    if (local instanceof LocalDdlMigrationResource) {
+    if (local instanceof LocalUriMigrationResource) {
+      script = convertScript(local.content());
+      checksum = ((LocalUriMigrationResource)local).checksum();
```

Comment:
I think we must change the order here. (first checksum, then convert)

Yes, this may break things, and some users will be afraid, but I think, the current implementation is not reliable, if parameters are involved that will change on runtime.

Maybe we need a legacy option to restore the old behaviour without reading the idx file (or taking the checksum)

Reply:
> (first checksum, then convert)

Yes, I believe that is what we want to get to. It's a question of how to do it with least impact etc. It looks like we can compute the checksum before and after (have 2 checksums) and then check **_both_** of them and automatically patch if there is a match on the old checksum.